### PR TITLE
Fixes Security Rockets

### PR DESCRIPTION
**Pull request by bathosbathtime**
[bugfix] [tweak]

This PR fixes EMP, stun, and blank rockets.
I was probably unnecessarily hostile to Sonix in the first OP and he still helped me fix the rockets so I would like to thank him. Further, the PR now also fixes low yield rocket damage as well as the stun time on nuke ops rockets. Ops rockets gib all but the most armored targets but this still changes behavior so it should be mentioned. To be clear, this is just fixing the rockets to apply the damage and stun values before they explode.

:cl:
 * bugfix: EMP, stun, and blank rockets now actually work.
 * bugfix: Low yield rockets AND nuclear ops rockets now properly apply damage and stun on direct hits.

### CODE REVIEW

**Comment by tuypo1** [comment]
I see no reason to remove xtreme rockets

**Comment by bathosbathtime** [comment]
I guess there isn't, I just find their inclusion to be odd. Un-removed.

**Comment by soniixapache** [comment]
All of this can be found in commit 431a7239329aa3abc16530567a0645e9be7b9219

>This PR fixes EMP, stun, and blank rockets. When sonix "cleaned up" the code he removed functionality on the EMP and stun rockets on accident and seemingly removed functionality on the blank rockets on purpose. What an ass.

I didn't do anything on purpose. I did accidentally break stun and EMP rockets. Thanks for fixing that.

>Oh he also snuck in an admin only "extreme" version of the low yield rocket in his code cleanup as well which I have now left in instead of removing them as I planned as there is really no reason to remove them.

I didn't 'sneak' anything, it was in the PR body. Here's a picture of it, since you can't see it for now.
rocket


>It may look like I removed the damage from the low yield rockets, however, that damage doesn't connect as the rocket explodes AS it bumps into something. This is also part of the reason why the other rockets had no effect. The other reason is that he named the to_bumps incorrectly.

Far as I was aware, the 'damage' var was applied to anyone that the rocket directly impacted, before applying the explosion. If that is not the case, it doesn't really matter either way. The *only* reason the stun and emp rockets don't work is that I named the `to_bump`s incorrectly.

>Also because he straight up removed the stun on the blank rockets.
In his "code cleanup."

Wrong. You don't seem to understand how inheritance works. Blank rockets inherit their stun var from the base class. They also inherit the explosion proc from the base class, so that part is unnecessary.


No need to be so hostile. The cleanup removed a lot of copypaste.

File: code/modules/projectiles/projectile/rocket.dm [hunk 1]
```diff
@@ -56,35 +53,48 @@
...
+	stun = 5
 	weaken = 10
 	agony = 10
-	exdev 	= -1
-	exheavy = 0
-	exlight = 0
-	exflash = 0
+	explosive = 0
+
+/obj/item/projectile/rocket/blank/to_bump(var/atom/A)
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, entire proc is inherited - just don't delete the ex- vars and it'll work™

File: code/modules/projectiles/projectile/rocket.dm [hunk 2]
```diff
@@ -56,35 +53,48 @@
 /obj/item/projectile/rocket/blank
 	name = "blank rocket"
 	damage = 5
+	stun = 5
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, inherited

File: code/modules/projectiles/projectile/rocket.dm [hunk 3]
```diff
@@ -56,35 +53,48 @@
...
 	damage = 10
 	agony = 30
 	emheavy = 3
 	emlight = 5
+	explosive = 0
 
-/obj/item/projectile/rocket/emp/to_bump(var/atom/A)
+/obj/item/projectile/rocket/blank/emp/to_bump(var/atom/A)
 	empulse(A, 3, 5)
+	explosion(A, -1, 0, 0, 0)
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, inherited from `..()`

File: code/modules/projectiles/projectile/rocket.dm [hunk 4]
```diff
@@ -56,35 +53,48 @@
...
 	..()
-	
+	if(!gcDestroyed)
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, inherited from `..()`

File: code/modules/projectiles/projectile/rocket.dm [hunk 5]
```diff
@@ -56,35 +53,48 @@
...
 	stun = 20
 	weaken = 20
 	agony = 30
+	explosive = 0
+
 
-/obj/item/projectile/rocket/stun/to_bump(var/atom/A)
+/obj/item/projectile/rocket/blank/stun/to_bump(var/atom/A)
 	flashbangprime(TRUE, FALSE, FALSE)
+	explosion(A, -1, 0, 0, 0)
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, inherited from `..()`

File: code/modules/projectiles/projectile/rocket.dm [hunk 6]
```diff
@@ -56,35 +53,48 @@
...
 	..()
-		
+	if(!gcDestroyed)
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: unnecessary, inherited from `..()`

**Review by soniixapache — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/rocket.dm [hunk 7]
```diff
@@ -45,9 +45,6 @@
 /obj/item/projectile/rocket/lowyield
 	name = "low yield rocket"
 	icon_state = "rpground"
-	damage = 45
```
Comment (soniixapache) [inline code/modules/projectiles/projectile/rocket.dm]: why are you removing these vars from only this subtype when both parent and other children have the var?

**Comment by Kurfursten** [comment]
Fix it

File: code/modules/projectiles/projectile/rocket.dm (hunk 2 above)
Comment (bathosbathtime) [reply code/modules/projectiles/projectile/rocket.dm]: Removed, sorry about that

File: code/modules/projectiles/projectile/rocket.dm (hunk 7 above)
Comment (bathosbathtime) [reply code/modules/projectiles/projectile/rocket.dm]: Reinstated, sorry about that

**Comment by bathosbathtime** [comment]
Applied the fix Sonix came up with, worked in testing. Still need to do some light clean up and more testing though. Should be done with it tonight.

File: code/modules/projectiles/projectile/rocket.dm (hunk 5 above)
Comment (bathosbathtime) [reply code/modules/projectiles/projectile/rocket.dm]: removed

**Comment by bathosbathtime** [comment]
Should be good to go, all rockets worked as expected when fired into a human mob and into a wall.

**Pull request merged**